src/mqtt_client.rs

```diff
@@ -5,27 +5,35 @@ use crate::{
     Property, {debug, error, info},
 };
 
-use core::{cell::RefCell, str::FromStr};
+use core::{cell::RefCell, convert::TryFrom, str::FromStr};
 
 use embedded_nal::{nb, IpAddr, Mode, SocketAddr};
+use embedded_time::{duration::Seconds, fixed_point::FixedPoint, Clock, Instant};
 
 use generic_array::{ArrayLength, GenericArray};
 use heapless::String;
 
 /// A client for interacting with an MQTT Broker.
-pub struct MqttClient<T, N>
+pub struct MqttClient<T, N, C>
 where
     T: ArrayLength<u8>,
     N: embedded_nal::TcpStack,
+    C: Clock,
+    u32: TryFrom<C::T>,
 {
     /// The network stack originally provided to the client.
     pub network_stack: N,
+    pub clock: C,
```

Comment:
If we keep the clock, I don't think it needs to be public.

Reply:
By the same token, the network stack wouldn't need to be public, though?